test/TestGrains/SiloAddressUtils.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UnitTests.Grains
+{
+    public static class SiloAddressUtils
```

Comment:
TestExtensions is probably a better place for this, isn't it?
Also, do you really need to use a real IP address with all the complex logic it brings? can't you just use the loopback or another hardcoded address for the tests?

Reply:
The issue is that TestExtensions depends on OrleansTestingHost, not sure we want to add that dependencies on test grain interfaces project.

I already simplified a bit the logic, but you are probably right I could just return the loopback for tests

Reply:
ah, I didn't realize this was being used by grain themselves. Well, if not needing a real IP Address is valid, then you won't need this utility at all